pkl-core/src/main/java/org/pkl/core/module/ModuleKeys.java

```diff
@@ -316,7 +316,13 @@ public List<PathElement> listElements(SecurityManager securityManager, URI baseU
     public ResolvedModuleKey resolve(SecurityManager securityManager)
         throws IOException, SecurityManagerException {
       securityManager.checkResolveModule(uri);
-      var realPath = path.toRealPath();
+      // Disallow paths that contain `\\` characters if on Windows
+      // (require `/` as the path separator on all OSes)
+      var uriPath = uri.getPath();
+      if (java.io.File.separatorChar == '\\' && uriPath != null && uriPath.contains("\\")) {
+        throw new FileNotFoundException();
```

Comment:
Fair point; will add an extra hint here about it. But I'm thinking let's just show this on every OS. It may or may not be useful to users on *nix platforms that are used to Windows style paths. Also, having this be Windows only greatly complicates our testing.